scripts/publish.js

```diff
@@ -0,0 +1,265 @@
+/**
+ * @fileoverview Publishes all of the packages in dependency order
+ * via GitHub workflow.
+ *
+ * Usage:
+ *
+ *   node scripts/publish.js [--dry-run]
+ *
+ * @author Nicholas C. Zakas
+ */
+
+//-----------------------------------------------------------------------------
+// Imports
+//-----------------------------------------------------------------------------
+
+import {
+	getPackageDirs,
+	calculatePackageDependencies,
+	createBuildOrder,
+} from "./shared.js";
+import { execSync } from "node:child_process";
+import { existsSync, readFileSync } from "node:fs";
+import { join } from "node:path";
+
+//-----------------------------------------------------------------------------
+// Read CLI Args
+//-----------------------------------------------------------------------------
+
+const dryRun = process.argv.includes("--dry-run");
+
+// for dry runs only output to console and don't execute anything
+const exec = dryRun ? text => console.log(text) : execSync;
+
+//-----------------------------------------------------------------------------
+// Helpers
+//-----------------------------------------------------------------------------
+
+/**
+ * Converts a GitHub Actions step output name to its corresponding environment variable name
```

Comment:
Do GitHub Actions automatically set environment variables for step outputs? I couldn't find any docs about this. I tried the following and it didn't work:

```yml
name: Test Steps Output
on: workflow_dispatch
jobs:
    test:
        runs-on: ubuntu-latest
        steps:
            - id: release
              run: echo "packages/config-array--release_created=true" >> "$GITHUB_OUTPUT"

            # prints "true"  
            - run: echo result from steps ${{ steps.release.outputs['packages/config-array--release_created'] }}            

            # prints nothing
            - run: echo result from env $STEPS_RELEASE_OUTPUTS_PACKAGES__CONFIG_ARRAY__RELEASE_CREATED

            # no env variable there
            - run: export
```

Reply:
Ah, I thought you might be running locally. :smile: 

It looks like my AI companion got this one wrong, I'll dig in.